File: DeliveryCommerce/app/src/main/java/lib/CustomTimer.java
package lib;




import android.os.Handler;
import android.util.Log;


public class CustomTimer
{
	
	
	private Handler tHandler;
	private Runnable tRunnable;
	private TimerDelegate delegate;
	private int repeat,count,time;
	
	public CustomTimer(int _time,int _repeat,TimerDelegate _delegate) {
		
		count=0; 
		delegate=_delegate;
		repeat=_repeat;
		time=_time;
		
	}
	
	public void timerStart(){
		timerStop();
		tHandler=new Handler();
		final CustomTimer that=this;
		tRunnable=new Runnable(){
	    	 public void run() {   
	    		 count++;
	    		 delegate.onTime(that);
	    		 if(count==repeat){
	    			 delegate.onComplete(that);
	    			 timerStop();
	    		 }else{
	    			 if(tHandler!=null){
	    			    tHandler.postDelayed(tRunnable, time);
	    			 }
	    		 }
            };
	     };
	     
	     tHandler.postDelayed(tRunnable, time);
	}
	public void timerStop(){
		if(tHandler!=null){
			tHandler.removeCallbacks(tRunnable);
		} 
	    tHandler=null;
		tRunnable=null;
	}
	public void resetTimer(){
		count=0;
		timerStop();
	}
	public void removeTimer(){
		timerStop();
		delegate=null;
	
	}
	public interface TimerDelegate
	{
	    void onTime(CustomTimer timer);
	    void onComplete(CustomTimer timer);
	   
	}
}

File: Magellan/app/src/main/java/lib/CustomTimer.kt
package lib

import android.os.Handler

class CustomTimer(private var time: Int, private val repeat: Int, private var delegate: TimerDelegate?) {


    private var tHandler: Handler? = null
    private var tRunnable: Runnable? = null
    var count: Int = 0

    fun timerStart()
    {
        timerStop()
        tHandler = Handler()
        val that = this
        tRunnable = Runnable {
            count++
            delegate?.onTime(that)
            if (count == repeat)
            {
                delegate?.onComplete(that)
                timerStop()
            }
            else
            {
                tHandler?.postDelayed(tRunnable, time.toLong())
            }
        }
        tHandler?.postDelayed(tRunnable, time.toLong())
    }

    fun timerStop()
    {
        tHandler?.removeCallbacks(tRunnable)
        tHandler = null
        tRunnable = null
    }

    fun resetTimer(t: Int)
    {
        count = 0
        time = t
        timerStop()
    }

    fun resetTimer()
    {
        count = 0
        timerStop()
    }

    fun removeTimer()
    {
        timerStop()
        delegate = null
    }

    interface TimerDelegate
    {
        fun onTime(timer: CustomTimer){}
        fun onComplete(timer: CustomTimer){}
    }
}